src/main/java/uk/ac/cam/cl/dtg/isaac/api/QuizFacade.java

```diff
@@ -0,0 +1,192 @@
+        this.contentManager = contentManager;
+
+        this.quizManager = quizManager;
+        this.userManager = userManager;
+    }
+
+    /**
+     * Get quizzes visible to this user.
+     *
+     * Anonymous users can't see quizzes.
+     * Students can see quizzes with the visibleToStudents flag set.
+     * Teachers and higher can see all quizzes.
+     *
+     * @return a Response containing a list of ContentSummaryDTO for the visible quizzes.
+     */
+    @GET
+    @Path("available")
+    @Produces(MediaType.APPLICATION_JSON)
+    @GZIP
+    @ApiOperation(value = "Get quizzes visible to this user.")
+    public final Response getAvailableQuizzes(@Context final HttpServletRequest request) {
+        try {
+            AbstractSegueUserDTO user = this.userManager.getCurrentUser(request);
+
+            if (!isUserRegistered(user)) {
+                throw new ForbiddenException("Only registered users can see quizzes.");
+            }
+
+            boolean isStudent = !isUserTeacherOrAbove(user);
+
+            EntityTag etag = new EntityTag(this.contentManager.getCurrentContentSHA().hashCode() + "");
+
+            ResultsWrapper<ContentSummaryDTO> summary = this.quizManager.getAvailableQuizzes(isStudent, null, null);
+
+            return Response.ok(summary).tag(etag)
+                .cacheControl(getCacheControl(NUMBER_SECONDS_IN_ONE_HOUR, isStudent))
+                .build();
+        } catch (SegueDatabaseException e) {
+            String message = "SegueDatabaseException whilst getting available quizzes";
+            log.error(message, e);
+            return new SegueErrorResponse(Response.Status.INTERNAL_SERVER_ERROR, message).toResponse();
+        } catch (ContentManagerException e) {
+            String message = "ContentManagerException whilst getting available quizzes";
+            log.error(message, e);
+            return new SegueErrorResponse(Response.Status.INTERNAL_SERVER_ERROR, message).toResponse();
+        }
+    }
+
+    /**
+     * Preview a quiz. Only available to teachers and above.
+     *
+     * @param request
+     *            - so we can deal with caching.
+     * @param httpServletRequest
+     *            - so that we can extract user information.
+     * @param quizId
+     *            as a string
+     * @return a Response containing a list of ContentSummaryDTO for the visible quizzes.
+     */
+    @GET
+    @Path("/{quizId}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @GZIP
+    @ApiOperation(value = "Preview an individual quiz.")
+    public final Response previewQuiz(@Context final Request request, @Context final HttpServletRequest httpServletRequest,
+                                      @PathParam("quizId") final String quizId) {
+        try {
+            AbstractSegueUserDTO user = this.userManager.getCurrentUser(httpServletRequest);
+
+            if (!(isUserTeacherOrAbove(user))) {
+                throw new ForbiddenException("Students cannot preview quizzes.");
```

Comment:
We tend to use `return SegueErrorResponse.getIncorrectRoleResponse();` rather than whatever this  `ForbiddenException` does.